File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
import collections


def solution(participant, completion):
    answer = collections.Counter(participant) - collections.Counter(completion)
    return list(answer.keys())[0]

# from collections import defaultdict

# def solution(participant, completion):
#     answer = ''
#     finish = defaultdict(int)
#     for i in participant:
#         finish[i] += 1
#     for i in completion:
#         finish[i] -= 1
#     for i in finish:
#         if finish[i] >= 1:
#             answer += i
        
#     return answer

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    
    let par_arr = participant.reduce((arr,val)=>{
       arr[val] = (arr[val] || 0) +1;
        return arr;
    },{})
    
    completion.map(c =>{
        par_arr[c] -= 1;
    })
    
    for(let i in par_arr){
        if(par_arr[i]){
            answer += i;
        }
    }
    
    
    return answer;
}